step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackagePlugin.java

```diff
@@ -77,11 +81,22 @@ public void afterInitializeData(GlobalContext context) throws Exception {
 
         packageExecutor = new AutomationPackageExecutor(
                 context.getScheduler(),
-                context.get(FunctionManager.class),
-                context.get(FunctionAccessor.class),
-                context.get(IsolatedAutomationPackageRepository.class)
+                context.require(FunctionManager.class),
+                context.require(FunctionAccessor.class),
+                context.require(IsolatedAutomationPackageRepository.class)
         );
         context.put(AutomationPackageExecutor.class, packageExecutor);
 
     }
+
+    @Override
+    public void serverStop(GlobalContext context) {
+        super.serverStop(context);
+        try {
+            this.packageExecutor.shutdown();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();  //set the flag back to true
```

Comment:
@jeromecomte 
Handling interruption is always painful for me... 

The idea was to protect from cases, when the calling code continues the execution after interruption. Like in simple example below, where the `for` loop will still be continued after the JVM receives the interruption signal (in `MyRunnable` we don't pass the interruption to the calling code and `sleep` is executed, because the thread doesn't have the 'interrupted' flag set).

`class Scratch {
    public static void main(String[] args) throws InterruptedException {
        Runnable runnable = new MyRunnable();
        for (int i = 0; i < 1000; i++) {
            runnable.run();
            Thread.sleep(100);
        }
    }

    public static class MyRunnable implements Runnable {

        @Override
        public void run() {
            try {
                someInterruptableMethod();
            } catch (InterruptedException e) {
                // can be used to pass interruption to the calling code and break the loop on Thread.sleep()
                // Thread.currentThread().interrupt();
            }
        }

        public void someInterruptableMethod() throws InterruptedException {
            // some logic that can throw InterruptedException
        }
    }
}`

But you are right, in our case the `Thread.currentThread().interrupt()` looks unnecessary here. I'll remove it.